.github/workflows/demo.yml

```diff
@@ -0,0 +1,123 @@
+name: Pulp Python 101
+
+on:
+  schedule:
+    - cron: '30 2 * * 0'
+
+env:
+  COLORTERM: 'yes'
+  PY_COLORS: '1'
+  TERM: 'xterm-256color'
+
+jobs:
+  demo:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/setup-python@v2
+        with:
+          python-version: "3.8"
+
+      - name: Starting Pulp server
+        run: |
+            mkdir settings pulp_storage pgsql containers
+            echo "CONTENT_ORIGIN='https://pulp.example.com'" >> settings/settings.py
+            docker run --detach \
+                      --name pulp \
+                      --publish 443:443 \
+                      --volume "/$(pwd)/settings:/etc/pulp:Z" \
+                      --volume "/$(pwd)/pulp_storage":/var/lib/pulp:Z \
+                      --volume "/$(pwd)/pgsql":/var/lib/pgsql:Z \
+                      --volume "/$(pwd)/containers":/var/lib/containers:Z \
+                      --device /dev/fuse \
+                      pulp/pulp:https
+
+            # Wait server to be up
+            sleep 10
+            for _ in $(seq 20)
+            do
+              sleep 2
+              if curl --insecure --fail https://localhost/pulp/api/v3/status/ > /dev/null 2>&1
+              then
+                break
+              fi
+            done
+
+            # Edit /etc/hosts
+            echo 127.0.0.1 pulp.example.com | sudo tee -a /etc/hosts
+
+            # Hack: adding pulp CA to certifi.where()
+            sudo docker cp pulp:/etc/pulp/certs/pulp_webserver.crt /usr/local/share/ca-certificates/pulp_webserver.crt
+            pip install certifi
+            CERTIFI=$(python -c 'import certifi; print(certifi.where())')
+            cat /usr/local/share/ca-certificates/pulp_webserver.crt | tee -a $CERTIFI
+
+            # Hack: adding pulp CA to default CA file
+            CERT=$(python -c 'import ssl; print(ssl.get_default_verify_paths().openssl_cafile)')
+            cat $CERTIFI | sudo tee -a $CERT
+            pip config set global.cert $CERT
+
+            # Updating certs
+            sudo update-ca-certificates
+
+            # Setting the password for admin
+            docker exec pulp bash -c 'pulpcore-manager reset-admin-password --password ci_password'
+
+      - name: Setting Pulp CLI
+        run: |
+          pip install pulp-cli[pygments]
+          pulp config create --username admin --base-url https://pulp.example.com --password ci_password
+          pulp status
```

Comment:
Starting from here, these are the client steps.

Let's split the client steps into 3 types of users:
- admin
- regular user
- QE

Here we are setting the CLI for the admin, the person who will create remotes/repositories/publications/distributions/...